src/logger/logger.js

```diff
@@ -30,16 +30,21 @@ export class Logger {
 		}
 		return false;
 	}
-	fmtMessage(level, msg, args) {
+	fmtMessage(level, msg, args = []) {
 		if (msg instanceof Error) {
 			msg = errToStr(msg);
 		}
+		if (typeof msg === 'object') {
+			args.shift(msg);
+			msg = '%2j';
+		}
 		args = args.map(e => (e instanceof Error ? errToStr(e) : e));
 		if (typeof msg === 'string') {
 			try {
 				msg = vsprintf(msg, args);
 			} catch (error) {
 				this.error(error);
+				this.error('could not format msg ' + msg);
```

Comment:
We should prefer using template strings where possible.

Reply:
fixing